packages/graphql-hooks/test/unit/useClientRequest.test.js

```diff
@@ -68,6 +68,30 @@ describe('useClientRequest', () => {
     )
   })
 
+  it('resets data when reset function is called', async () => {
+    let fetchData, state, resetData
+    renderHook(
+      () => ([fetchData, state, resetData] = useClientRequest(TEST_QUERY)),
+      {
+        wrapper: Wrapper
+      }
+    )
+    // initial state
+    expect(state).toEqual({ cacheHit: false, loading: true })
+    await act(fetchData)
+    expect(state).toEqual({
+      cacheHit: false,
+      loading: false,
+      data: 'data'
+    })
+    await act(resetData)
+    // should be back to initial state
+    expect(state).toEqual({
```

Comment:
I'm wondering if there's a case where falling back to the `initialState` completely is not preferable. 
The issue mentions only rolling back the value of `data` in the `state`. Is there a use case where we might only want to roll back the `data` value?

Reply:
@jh3y I think it is related to these:
https://github.com/apollographql/apollo-feature-requests/issues/157
https://github.com/apollographql/apollo-feature-requests/issues/170

Reply:
What if we provided a way to pass in a "desired" state that defaults to initial?
That way if a user wants to preserve part of the state, but not all of it, they can.

`resetState({...currentState, data: undefined})`

Or

`resetState()`

Reply:
Sounds good, I've now added it, have a look and let me know what you think.

Reply:
Looks good! 👍